docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,958 @@
+| `FailureMessage` (deprecated)        | `BackCompatibilty.FailureMessage` (renamed) (deprecated)    | (removed)                           |
+| `Conditions`                         | `BackCompatibilty.Conditions` (renamed) (deprecated)        | (removed)                           |
+| `ExperimentalConditions` (new)       | `Conditions` (renamed)                                      | `Conditions`                        |
+| other fields...                      | other fields...                                             | other fields...                     |
+
+Notes:
+- The `BackCompatibilty` struct is going to exist in v1beta2 types only until v1beta1 removal (9 months or 3 minor releases after v1beta2 is released/v1beta1 is deprecated, whichever is longer).
+  Fields in this struct are used for supporting down conversions, thus providing users relying on v1beta1 APIs additional buffer time to pick up the new changes.
+
+#### MachineDeployment (New)Conditions
+
+| Condition        | Note                                                                                                                                                                                                                                                   |
+|------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Available`      | True if the MachineDeployment has minimum availability according to parameters specified in the deployment strategy, e.g. If using RollingUpgrade strategy, availableReplicas must be greater or equal than desired replicas - MaxUnavailable replicas |  
+| `ReplicaFailure` | This condition surfaces issues on creating a MachineSet replica in Kubernetes, if any. e.g. due to resource quotas.                                                                                                                                    |
+| `MachinesReady`  | This condition surfaces detail of issues on the controlled machines, if any.                                                                                                                                                                           |
+| `ScalingUp`      | True if available replicas < desired replicas                                                                                                                                                                                                          |
+| `ScalingDown`    | True if replicas > desired replicas                                                                                                                                                                                                                    |
```

Comment:
If going through a rollout with something like `MaxSurge = 3`, is `ScalingDown` going to report `True`? (And vice versa with `MaxUnavailable` and `ScalingUp`?)

Reply:
If going through a rollout with something like MaxSurge = 3 and MaxUnaivailable=0, ScalingDown will not show up (it depends by both settings, not only by one).
Same for ScalingUp

Reply:
Hopefully now it is clear, resolving